### PR TITLE
HPCC-16431 Coverity: Resource leak WsDeployService.cpp

### PR DESCRIPTION
**Pull request by richardkchapman**
pNode would leak if pTopology was not found. Also cleaned up some silly code.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: esp/services/WsDeploy/WsDeployService.cpp [hunk 1]
```diff
@@ -4929,47 +4929,47 @@ bool CWsDeployFileInfo::handleTopology(IEspContext &context, IEspHandleTopologyR
...
+      if (!strcmp(sbNewType, XML_TAG_ECLCCSERVERPROCESS) ||
+        !strcmp(sbNewType, XML_TAG_ECLSERVERPROCESS) ||
+        !strcmp(sbNewType, XML_TAG_ECLAGENTPROCESS) ||
+        !strcmp(sbNewType, XML_TAG_ECLSCHEDULERPROCESS) ||
+        !strcmp(sbNewType, XML_TAG_ROXIECLUSTER))
+        pNode->addProp(XML_ATTR_PROCESS, "");
+      else if (!strcmp(sbNewType, XML_TAG_THORCLUSTER))
+      {
+        StringBuffer strName("thor");
+        strName.set(getUniqueName2(pEnvRoot->queryPropTree(buf.str()), strName, "ThorCluster", XML_ATTR_PROCESS));
```
Comment (ghalliday) [inline esp/services/WsDeploy/WsDeployService.cpp]: This line looks very dodgy - it is setting strName to the value which has been assigned to strName.  The strName.set() should be removed.

Comment (ghalliday) [inline esp/services/WsDeploy/WsDeployService.cpp]: (as it is assigned that may cause the value to be invalidated).

**Review by ghalliday — DISMISSED** [review]

**Comment by richardkchapman** [comment]
@ghalliday Repushed to address the comment

**Pull request merged**